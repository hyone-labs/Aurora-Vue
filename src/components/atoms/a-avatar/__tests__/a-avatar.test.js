import { render, screen } from '@testing-library/vue';
import AAvatar from '../a-avatar.vue';

describe('AAvatar', () => {
  it('render correctly with default props', () => {
    const { html } = render(AAvatar);
    expect(html()).toMatchSnapshot();
  });

  it('render correctly with gender', async () => {
    render(AAvatar, {
      props: {
        gender: 'female',
      },
    });
    expect(screen.getByRole('img').hasAttribute('src')).toBe(true);
    expect(screen.getByRole('img').getAttribute('src')).toBe('https://hyone-backend-homolog.herokuapp.com/public/images/undraw/avatar_female.svg');
  });

  it('render correctly with src', async () => {
    render(AAvatar, {
      props: {
        src: 'www.my-custom-url.com',
      },
    });
    expect(screen.getByRole('img').hasAttribute('src')).toBe(true);
    expect(screen.getByRole('img').getAttribute('src')).toBe('www.my-custom-url.com');
  });

  it('render correctly with type', async () => {
    render(AAvatar, {
      props: {
        type: 'png',
      },
    });
    expect(screen.getByRole('img').hasAttribute('src')).toBe(true);
    expect(screen.getByRole('img').getAttribute('src')).toBe('https://hyone-backend-homolog.herokuapp.com/public/images/undraw/avatar_male.png');
  });

  it('render correctly with size', async () => {
    render(AAvatar, {
      props: {
        size: 'medium',
      },
    });
    expect(screen.getByRole('figure').classList.contains('a-avatar--size-medium')).toBe(true);
  });

  it('render correctly with behavior', async () => {
    render(AAvatar, {
      props: {
        behavior: 'square',
      },
    });
    expect(screen.getByRole('figure').classList.contains('a-avatar--behavior-square')).toBe(true);
  });

  it('render correctly with caption', async () => {
    const { html } = render(AAvatar, {
      props: {
        caption: 'My caption',
      },
    });
    expect(html()).toMatchSnapshot();
    expect(screen.getByText(/my caption/i).textContent.trim()).toBe('My caption');
  });
});
