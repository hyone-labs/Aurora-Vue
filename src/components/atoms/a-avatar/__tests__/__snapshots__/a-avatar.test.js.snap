// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AAvatar render correctly with caption 1`] = `
"<figure class=\\"a-avatar a-avatar--size-colossal a-avatar--behavior-rounded\\"><img src=\\"https://hyone-backend-homolog.herokuapp.com/public/images/undraw/avatar_male.svg\\" class=\\"a-avatar__image\\">
  <figcaption class=\\"a-avatar__figcaption\\">
    My caption
  </figcaption>
</figure>"
`;

exports[`AAvatar render correctly with default props 1`] = `
"<figure class=\\"a-avatar a-avatar--size-colossal a-avatar--behavior-rounded\\"><img src=\\"https://hyone-backend-homolog.herokuapp.com/public/images/undraw/avatar_male.svg\\" class=\\"a-avatar__image\\">
  <!---->
</figure>"
`;
